src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -758,11 +758,14 @@ def send_raw_request(cli_ctx, method, url, headers=None, uri_parameters=None,  #
                         resource = value
                         break
         if resource:
-            # If this is an ARM request, extract subscription ID from the URL.
-            # In the future when multi-tenant subscription is supported, we won't be able to uniquely identity the token
-            # from subscription anymore.
+            # If this is an ARM request, try to extract subscription ID from the URL.
+            # But there are APIs which don't require subscription ID, like /subscriptions, /tenants
+            # TODO: In the future when multi-tenant subscription is supported, we won't be able to uniquely identity
+            #   the token from subscription anymore.
+            token_subscription = None
             if url.lower().startswith(endpoints.resource_manager.rstrip('/')):
                 token_subscription = _extract_subscription_id(url)
+            if token_subscription:
```

Comment:
>token_subscription [](start = 15, length = 18)

It seems the logic is changed:
before: if url doesn't start with "endpoints.resource_manager.rstrip('/')", it will go to "else" clause,
now: if  url starts with "endpoints.resource_manager.rstrip('/')", but cannot get token_subscription, it will also go to "else" clause, is it expected behavior?

Reply:
Yes. The `else` clause will be reached under 2 scenarios:

1. The URL doesn't start with `https://management.azure.com/`, such as `https://graph.microsoft.com/v1.0/applications`
2. The URL starts with `https://management.azure.com/`, but subscription ID can't be extracted, such as `https://management.azure.com/subscriptions?api-version=2020-01-01`

Reply:
this is the key change. right? Do I miss understand?

Reply:
Yes, this is where the change takes effect.